Core/GameInstance.cs

```diff
@@ -72,14 +72,18 @@ public GameInstance(IGame game, string gameDir, string name, IUser user, bool sc
             }
         }
 
-        public bool Valid
-        {
-            get
-            {
-                return game.GameInFolder(new DirectoryInfo(gameDir))
-                    && Version() != null;
-            }
-        }
+        /// <returns>
+        /// true if the game seems to be here and a version is found,
+        /// false otherwise
+        /// </returns>
+        public bool Valid => game.GameInFolder(new DirectoryInfo(gameDir)) && Version() != null;
+
+        /// <returns>
+        /// true if the instance may be locked, false otherwise.
+        /// Note that this is a tentative value; if it's true,
+        /// we still need to try to acquire the lock to confirm it isn't stale.
+        /// </returns>
+        public bool IsMaybeLocked => RegistryManager.IsInstanceMaybeLocked(CkanDir());
```

Comment:
If a KSP instance is invalid (or the NFS share it's on is not mounted, as in my case), `CkanDir()` throws a `NotKSPDirKraken`, which we should catch.
It's debatable whether it should return false or true in this case, but I'm leaning towards false, since I think I wouldn't expect the ManageGameInstancesDialog showing `(LOCKED)` here.
But maybe it should show `(INVALID)`? We could add another check to `UpdateInstancesList()` with a new string:
```csharp
 _manager.CurrentInstance != instance.Value && instance.Value.IsMaybeLocked
    ? string.Format(Properties.Resources.ManageGameInstancesLocked, instance.Key)
    : instance.Value.Valid
        ? instance.Key
        : string.Format(Properties.Resources.ManageGameInstancesInvalid, instance.Key),
```

Reply:
Now checking `Valid` first and labelling those instances `(INVALID)` if needed.

Reply:
I might still be a good idea to catch this exception, in case we ever use `IsMaybeLocked()` somewhere else, so we don't get surprised by it again. What do you think?

Reply:
I see your point, exceptions are annoying. But I'm not sure I like the idea of `IsMaybeLocked` returning true _or_ false for an invalid instance, since the lock won't even have been checked. It doesn't really lead to a safer coding pattern...

```csharp
    if (instance.IsMaybeLocked)
    {
        // Unsafe to use instance here if we return true for invalid
    }
    else
    {
        // Unsafe to use instance here if we return false for invalid
    }
```

Maybe longer-term we could replace `Valid` and `IsMaybeLocked` with a `Status` property returning what's wrong. Then the calling code would know the different possibilities and would be responsible for handling them.

```csharp
    enum InstanceStatus
    {
        Invalid,
        MaybeLocked,
        OK,
    };
```

Reply:
He, I also thought about a status enum. Something we can keep in mind.

What about a code comment at least, that instructs to check `Valid` first? If we make that the intended calling order, it would make sense to communicate that to the caller.

Reply:
Works for me. :+1: